src/Database/Connection.php

```diff
@@ -790,6 +799,33 @@ public function cacheMetadata($cache): void
         $this->_config['cacheMetadata'] = $cache;
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function setCacher(CacheInterface $cacher)
+    {
+        $this->cacher = $cacher;
+
+        return $this;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getCacher(): CacheInterface
+    {
+        if ($this->cacher === null) {
+            $configName = $this->_config['cacheMetadata'] ?? '_cake_model_';
+            if (!is_string($configName)) {
+                $configName = '_cake_model_';
+            }
+
+            $this->cacher = Cache::pool($configName);
```

Comment:
I think that   we should somehow not force this dependency, and maybe provide a null cacher implementation instead of defaulting in this way.

Reply:
If we remove this then we will need to come up with another way to set default cacher for cake apps.

Currently we just call `ConnectionManager` in bootstrap to setup db config arrays and the rest happens "magically".

Reply:
I can make `getCacher()` return null if no cacher is explicitly set and move the fetching to default cache using `Cache` back into `CachedCollection`.

Reply:
Would it be better if we raised an error if `cakephp/cache` isn't available?

Reply:
Done. I also removed `cakehp/cache` as dependency.